subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4202,13 +4202,15 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 int lwm2m_parse_peerinfo(char *url, struct sockaddr *addr, bool *use_dtls)
```

Comment:
Can you elaborate in commit message, why something like this is required then offloading is used, it is not obvious.

Reply:
CONFIG_NET_IPV6 and CONFIG_NET_IPV4 initialized hints.ai_family. If CONFIG_NET_SOCKETS_OFFLOAD does not do the same, getaddrinfo() (implemented by Nordic offload sockets) returns error -9:

> [00:00:11.420,410] <inf> net_lwm2m_engine: Parse url: coap://leshan.eclipseprojects.io
> [00:00:11.420,471] <err> net_lwm2m_engine: Unable to resolve address (-9)